scripts/coderabbit_retry/coderabbit_retry.py

```
@@ -0,0 +1,384 @@
+            timeout=SUBPROCESS_TIMEOUT_SECONDS,
+        )
+    except (OSError, subprocess.TimeoutExpired) as error:
+        LOGGER.warning(f"PR #{pr_number}: rate-limit check failed: {error}")
+        return None
+    if result.returncode != 0:
+        LOGGER.warning(f"PR #{pr_number}: rate-limit check exited {result.returncode}: {result.stderr.strip()}")
+        return None
+    try:
+        return json.loads(result.stdout)
+    except json.JSONDecodeError as error:
+        LOGGER.warning(f"PR #{pr_number}: failed to parse rate-limit check JSON: {error}")
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)